src/utils/common/common.ts

```diff
@@ -47,6 +47,61 @@ export const forkRefs =
     });
   };
 
+// String is of type XX:YY or XX.YY where XX is 0-24 and YY is 0-59
+const isValidTimeString = (timeStr: string) => {
+  if (timeStr.match(/^\d{1,2}.\d{2}$/) || timeStr.match(/^\d{1,2}:\d{2}$/)) {
+    const parts = timeStr.split(timeStr.includes('.') ? '.' : ':');
+    const hours = parseInt(parts[0], 10);
+    const minutes = parseInt(parts[1], 10);
+    if (hours >= 0 && hours < 25 && minutes >= 0 && minutes < 60) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+/**
+ * Contains logic to autocomplete a time string in certain scenarios:
+ *  - 1-2 characters which form a valid hour designation are autocompleted to full time: 14 --> 14.00 and 9 --> 9.00
+ *  - 4 characters which form a valid "military time" will be autocompleted to full time: 1400 --> 14.00
+ *  - A leading zero will be removed from hours in an otherwise valid time: 09.00 --> 9.00
+ *  - The colon : character will be replaced by the dot . character in an otherwise valid time: 12:00 --> 12.00
+ * @param timeStr Time input value
+ * @returns An autocompleted time string or `null` if no automatic completions could be perfomed
+ */
+export const autocompleteTimeString = (timeStr: string): string | null => {
+  const inputValInt = parseInt(timeStr, 10);
+
```

Comment:
Could there be a simple overall validity check in the beginning to avoid unnecessary if clause checks?

Reply:
Added some now